app/src/main/java/bg/zahov/app/data/model/state/ExerciseScreenData.kt

```diff
@@ -0,0 +1,55 @@
+package bg.zahov.app.data.model.state
+
+import bg.zahov.app.data.model.BodyPart
+import bg.zahov.app.data.model.Category
+import bg.zahov.app.data.model.FilterItem
+
+/**
+ * UI state representation for the exercise view, including a list of exercises,
+ * filters, loading state, search query, current action flag, and dialog visibility.
+ *
+ * @property exercises The list of exercises available in the UI.
+ * @property filters The current applied filters.
+ * @property loading Indicates whether data is currently loading.
+ * @property search The current search query.
+ * @property flag The current action state for exercises (e.g., Adding, Replacing).
+ * @property showDialog Indicates whether to show a dialog.
+ */
+data class ExerciseScreenData(
```

Comment:
You can use objects to represent the state here.

Example:
```
sealed class ExerciseScreenUiState {
    data object NavigateBack : ExerciseScreenUiState()
    data object NavigateInfo : ExerciseScreenUiState()
    data class Exercises(val exercises: List<ExerciseData>) : ExerciseScreenUiState()
}
```
And handle it in Compose code like:
```
when(uiState) {
  is ExerciseScreenUiState.Exercises -> {}
  ...
}
```

Reply:
@dimitarstoyanovtumba I think as we agreed at the beginning navigation events and view model events should result in ui state https://developer.android.com/topic/architecture/ui-layer/events#handle-viewmodel-events and https://developer.android.com/topic/architecture/ui-layer/events#navigation-events

I only agree since Loading is a separate ui component it can be separate state

Reply:
Thanks @trifonov-i , I must've missed that detail. If that is the agreed approach I'm okay with it.